### PR TITLE
Update parsing to strip whitespace from table cells

### PR DESCRIPTION
**Pull request by alinab**

### CODE REVIEW

**Comment by alexbiehl** [comment]
Thanks Alina! Can you also add a test case to the fixture test in haddock-library?

**Comment by alinab** [comment]
I just added a test case for this as well updated fixture data for tables in 6 other files that were throwing up errors because of this update.

File: haddock-library/fixtures/examples/table-simple.parsed [hunk 1]
```diff
@@ -3,50 +3,44 @@ DocTable
...
-                                                  (DocAppend
-                                                     (DocMonospaced (DocString "OK"))
-                                                     (DocString "       ")),
+                                                  (DocMonospaced (DocString "OK")),
                             tableCellRowspan = 1},
                          TableCell
                            {tableCellColspan = 1,
-                            tableCellContents = DocString
-                                                  " operation successful                     ",
+                            tableCellContents = DocString " operation successful",
```
Comment (alexbiehl) [inline haddock-library/fixtures/examples/table-simple.parsed]: I wonder if we should strip leading whitespace as well?

**Comment by alexbiehl** [comment]
> I just added a test case for this as well updated fixture data for tables in 6 other files that were throwing up errors because of this update.

Awesome, thank you!

File: haddock-library/fixtures/examples/table-simple.parsed (hunk 1 above)
Comment (alinab) [reply haddock-library/fixtures/examples/table-simple.parsed]: I will look into that as a additional task.

Comment (alinab) [reply haddock-library/fixtures/examples/table-simple.parsed]: It involved a trivial change but I have added code that strips leading whitespace as well. Also test fixtures and one html-test reference have been updated for this.

**Comment by alexbiehl** [comment]
Thanks again Alina!

**Pull request merged**